base/lthooks.dtx

```diff
@@ -2195,6 +2195,15 @@
 %    \end{macrocode}
 % \end{macro}
 %
+% \begin{macro}{\@@_prop_gclear:c}
+%   Same as above.
+% \changes{v1.0u}{2022/05/13}{Add non-debug version of \cs{prop_gclear:c}}
+%    \begin{macrocode}
+\cs_new_protected:Npn \@@_prop_gclear:c #1
+  { \exp_args:Nc \@@_tl_gset_eq:NN {#1} \c_empty_prop }
```

Comment:
Yeah, `\prop_gclear_new:N` would do here too. The annoyance is that when `l3debug` is active, it would log `Defining \g__hook_<name>_code_prop on line ...` for every generic hook.  I don't mind much, but I think it'd be better not to clutter, if possible.

Reply:
the question is, should this be considered clutter? In some sense it seem right to show at which point a prop springs into existence if you should that at all and not show it for some but not for all of them. But I'm not too worried either way either.

Reply:
Maybe not clutter, but "misleading debug information".  The right debug info to show here would be something like `Clearing one-time hook '#1' data structure`.

`\prop_gclear_new:c` seems wrong here, as we don't really want to create a new prop.  In fact, I think unconditionally clearing it (and as a consequence creating a new prop) is a bit wasteful, and (I think) could instead be:
```latex
\prop_if_exist:cT { g_@@_#1_code_prop }
  { \prop_gclear:c { g_@@_#1_code_prop } }
```
I'm leaning toward this latter option, if it doesn't break anything...

Reply:
Perhaps try that then, might be the best.